wondrous-app/components/CreateEntity/CreateEntityModal/TaskTemplatePicker/index.tsx

```diff
@@ -0,0 +1,343 @@
+                    {template?.reviewer?.map((reviewer, index) => {
+                      if (index < template.reviewer.length - 1) {
+                        return <TaskTemplateReviewerValue>{`${reviewer.label}, `}</TaskTemplateReviewerValue>;
+                      } else {
+                        return <TaskTemplateReviewerValue>{`${reviewer.label}`}</TaskTemplateReviewerValue>;
+                      }
+                    })}
+                  </Grid>
+                  <TaskTemplateLabelValue>{`Assignee: ${template?.assignee?.label}`}</TaskTemplateLabelValue> 
+                  
+                  */}
+
+                  <TaskTemplateRewardContainer>
+                    {template?.rewards?.[0] ? (
+                      <TaskTemplateRewardBox>
+                        <TaskTemplateEthereumIcon />
+                        <TaskTemplateRewardValue>
+                          {template?.rewards?.[0]?.rewardAmount}{' '}
+                          {getPaymentMethodData(template?.rewards?.[0]?.paymentMethodId)?.symbol}
+                        </TaskTemplateRewardValue>
+                      </TaskTemplateRewardBox>
+                    ) : null}
+                    {template?.points ? (
+                      <TaskTemplateRewardBox>
+                        <CreateEntityDefaultDaoImage onClick={handleEllipsesClick} />
+
+                        <TaskTemplatePointsValue>{template?.points} points</TaskTemplatePointsValue>
+                      </TaskTemplateRewardBox>
+                    ) : null}
+                  </TaskTemplateRewardContainer>
+                </TaskTemplateListItems>
+              </TaskTemplateContainer>
+            );
+          })
+        ) : (
+          <TaskTemplateLabelValue
+            style={{
+              fontSize: '15px',
+            }}
+          >
+            No templates yet!
+          </TaskTemplateLabelValue>
+        )}
+      </TaskTemplateActionPopper>
+    );
+  };
+
+  const renderContainer = () => {
+    switch (saveOrOpen) {
+      case 'open':
+        return <TaskTemplateOpen />;
+      default:
+        return (
+          <TaskTemplateDefaultPopper
+            open={open}
+            anchorEl={anchorEl}
+            placement="top-start"
+            disablePortal={true}
+            style={sizingForPopperContainer()}
+          >
+            <TaskTemplateOptionsLabel
+              onClick={() => {
+                setSaveOrOpen('save');
+              }}
+            >
+              Save new template
+            </TaskTemplateOptionsLabel>
+            <TaskTemplateOptionsLabel
+              onClick={() => {
+                setSaveOrOpen('open');
+              }}
+            >
+              Open existing template
+            </TaskTemplateOptionsLabel>
+          </TaskTemplateDefaultPopper>
+        );
+    }
+  };
+
+  const sizingForPopperContainer = () => {
+    switch (saveOrOpen) {
```

Comment:
optional: can we switch to guard clauses? 

e.g `if(saveOrOpen === SAVE) return {}`